sirepo/pkcli/service.py

```diff
@@ -52,24 +59,67 @@ def flower():
         ])
 
 
+def flask():
+    from sirepo import server
+
+    use_reloader = pkconfig.channel_in('dev')
+    app = server.init(use_reloader=use_reloader)
+    # avoid WARNING: Do not use the development server in a production environment.
+    app.env = 'development'
+    app.run(
+        host=cfg.ip,
+        port=cfg.port,
+        threaded=True,
+        use_reloader=use_reloader,
+    )
+
+
 def http():
-    """Starts Flask server in http mode.
+    """Starts the Flask server and job_supervisor.
 
     Used for development only.
-    """
-    from sirepo import server
 
-    with pkio.save_chdir(_run_dir()):
-        use_reloader = pkconfig.channel_in('dev')
-        app = server.init(use_reloader=use_reloader)
-        # avoid WARNING: Do not use the development server in a production environment.
-        app.env = 'development'
-        app.run(
-            host=cfg.ip,
-            port=cfg.port,
-            threaded=True,
-            use_reloader=use_reloader,
+    Args:
+        driver (string): The driver type to enable (one of local, docker, sbatch, nersc)
+        nersc_proxy (string): A proxy nersc can use to reach the supervisor
+        nersc_user(string): A nersc user ??
+        sbatch_host (string): A host to ssh into to start sbatch jobs
+    """
+    def _env(py_version):
+        e = os.environ.copy()
+        e.update(
+            PYENV_VERSION=py_version,
+            SIREPO_JOB_DRIVER_MODULES='local',
+            SIREPO_MPI_CORES='2',
         )
+        return e
+
+    def _exit(*args):
+        for p in processes:
+            os.killpg(os.getpgid(p.pid), args[0])
+            p.wait()
+        sys.exit()
```

Comment:
Only exit this way if we get a signal. Normal exit should pass through normal cases.

In the case of Control-C we should exit with a non-zero code (1 is fine).